Cabal/src/Distribution/Simple/GHC/Build/Utils.hs

```diff
@@ -74,17 +74,21 @@ withDynFLib flib =
     ForeignLibTypeUnknown ->
       cabalBug "unknown foreign lib type"
 
+-- | Is the extension of the file in the list of extensions?
+extensionIn :: [String] -> FilePath -> Bool
+extensionIn exts fp = takeExtension fp `elem` exts
```

Comment:
`in` is usually interpreted as the set membership operator ∈, which implies the set argument on the left rather than on the right. So, the parameters of `extensionIn` should be swapped I believe.

Reply:
Of course I got the sides wrong... But the point stands.

Reply:
I put `[String]` as first argument to facilitate partial application.